submissions/api.py

```diff
@@ -48,64 +52,122 @@
 TOP_SUBMISSIONS_CACHE_TIMEOUT = 300
 
 
-def create_submission(student_item_dict, answer, submitted_at=None, attempt_number=None, team_submission=None):
-    """Creates a submission for assessment.
+def create_external_grader_detail(submission, event_data):
+    """
+    Creates a ExternalGraderDetail for a given submission.
+
+    Args:
+        submission (Submission): The submission object to create a queue record for.
+        event_data (dict): Data to be included in the queue record. Must include a 'queue_name' key.
+
+    Returns:
+        ExternalGraderDetail: The created queue record.
+
+    Raises:
+        ExternalGraderQueueCanNotBeEmptyError: If event_data doesn't contain required queue_name.
+        SubmissionInternalError: If there's an error creating the queue record.
+    """
+
+    if not event_data.get('queue_name'):
+        raise ExternalGraderQueueCanNotBeEmptyError("event_data must contain 'queue_name'")
+
+    try:
+        queue_record = ExternalGraderDetail.objects.create(
+            submission=submission,
+            queue_name=event_data['queue_name'],
+            grader_file_name=event_data.get('grader_file_name', ''),
+            points_possible=event_data.get('points_possible', 1),
```

Comment:
Should these be optional params to the function as well? What else goes in `event_data`?